x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/ltr/LearningToRankRescorer.java

```diff
@@ -56,22 +56,12 @@ public TopDocs rescore(TopDocs topDocs, IndexSearcher searcher, RescoreContext r
             throw new IllegalStateException("local model reference is null, missing rewriteAndFetch before rescore phase?");
         }
 
-        if (rescoreContext.getWindowSize() < topDocs.scoreDocs.length) {
```

Comment:
ℹ️ This case does not happen anymore since the top doc is truncated to the windows length by the rescore phase before we reach this point.

Reply:
> This case does not happen anymore since the topDocs is truncated to the windows length by the rescore phase before we reach this point.

I am unsure about that truncating. Seems very trappy.

Reply:
With the latest commit we'll fail the request if from+size is greater than windowSize so it's not trappy anymore?